### DISCUSSION TITLE
Support for glossy/reflective materials like Vroid Hub?

### DISCUSSION THREAD

**Discussion by manugill** [pixiv/three-vrm#1471] [Help]
Hi! I was browsing Vroid Hub and noticed that on a few models with clothing like body suits, there were some examples that seem to reflect light when camera is moved. It really improves the perceived quality of the model in my opinion.

Example: https://hub.vroid.com/en/characters/872066437993506369/models/1211940754382081947

Is it possible to do this with three-vrm?

I remember taking a look at the clientside JS code on Vroid Hub when I came across this. I decrypted the .vrm files served there to see if they were special. And turned out they were as they were using a special material/texture that supported this.

I think the plugin (if i recall correctly) to support glossy material did not seem be included in the open source version of the library.

Would appreciate any input from Pixiv's end on this.

**Comment by 0b5vr** [comment]
This glossy expression is from matcap texture of MToon, so you can already achieve this using MToon.

By the way, reverse engineering the VRoid Hub is explicitly prohibited in the Terms of Use, Article 4. Please do not attempt to analyze our client-side JavaScript code.
https://policies.pixiv.net/en.html#vroidhub

**Reply by manugill** [reply]
Thanks for the reply! Would you be able to point me to an example if possible, or provide some more detailed pointers on how to achieve it?

I honestly haven't been able to achieve any sort of glossiness to any clothing, nor found any resources to help. I've tried the following:
- attempted to create a model with Vroid Studio with a glossy texture. Didnt seem possible.
- Imported it in Blender a played around for a few hours trying to understand the material's properties.
- Tried to understand how vroid hub is doing it.

Haven't had any luck.

PS apologies, didn't have any nefarious intentions, definitely not "reverse engineering" your product. Just wanted to learn how the glossiness was working. And it didn't seem possible with the open source version so thought I'd provide context.
